checkQC/qc_reporter.py

```diff
@@ -0,0 +1,133 @@
+import re
+import logging
+
+import checkQC.qc_checkers
+import checkQC.views
+from checkQC.qc_checkers.utils import handler2checker
+
+
+log = logging.getLogger(__name__)
+
+
+class QCReporter:
+    """
+    Class to generate QC reports based on QCData.
+    """
+    def __init__(
+        self,
+        configs,
+    ):
+        self.configs = configs
+
+    def gather_reports(
+        self,
+        qc_data,
+        use_closest_read_len=False,
+        downgrade_errors_for=[],
+    ):
+        exit_status = 0
+
+        config = self._select_configs(
+            qc_data,
+            use_closest_read_len,
+            downgrade_errors_for,
+        )
+        checker_configs = config["checkers"]
+
+        qc_reports = [
+            qc_report
+            for checker, checker_config in checker_configs.items()
+            for qc_report in getattr(checkQC.qc_checkers, handler2checker(checker))(
+                qc_data,
+                **checker_config
+            )
+        ]
+
+        if any(qc_report.type() == "error" for qc_report in qc_reports):
+            exit_status = 1
+
+        return exit_status, getattr(checkQC.views, config["view"])(
+            checker_configs,
+            qc_data,
+            qc_reports,
+        )
+
+    def _select_configs(
```

Comment:
I might be missing something obvious here, but why does this class need to re-implement these functions? Could we just use the Config class to find the closest read length and picking the right handler configs? Configs that are used to initialize an object of this class could already be "filtered".

Reply:
Good question! I felt the `Config` and `ConfigFactory` classes were a bit overkill for something that could in principle be done with a simple dictionary and that only needs to be done once . I wanted to see how it would look like without using such classes.

Maybe we can keep it for now and see what works best when we connect `QCReporter` to the web handler.

Reply:
Yeah, that is a valid point. 

That sounds good :+1: